frontend/src/pages/onboarding.tsx

```diff
@@ -2,140 +2,89 @@ import { ErrorAlert } from "@components/_shared/Alerts";
 import { SingInLayout } from "@components/_shared/SignInLayout";
 import Spinner from "@components/_shared/Spinner";
 import { Button } from "@components/ui/button";
+import type { InferGetServerSidePropsType } from "next";
 
 import InterestsSteps from "@components/onboarding-steps/InterestsSteps";
 import InviteUsersStep from "@components/onboarding-steps/InviteUsersStep";
 import OrganizationSelectionStep from "@components/onboarding-steps/OrganizationSelectionStep";
 import { OnboardingFormType } from "@schema/onboarding.schema";
 import type { GetServerSidePropsContext } from "next";
-import { getCsrfToken } from "next-auth/react";
+import { getCsrfToken, useSession } from "next-auth/react";
 import { NextSeo } from "next-seo";
 import { useRouter } from "next/router";
-import { ReactNode, useEffect, useState } from "react";
+import { ReactNode, useEffect, useState, Fragment } from "react";
 import { useForm } from "react-hook-form";
 import { match } from "ts-pattern";
+import { listGroups, followGroups } from "@utils/group";
+import { inviteUser } from "@utils/user";
+import { listOrganizations, requestOrganizationOwner, requestNewOrganization } from "@utils/organization";
 
 export async function getServerSideProps(context: GetServerSidePropsContext) {
+  const csrfToken = await getCsrfToken(context);
+  
+  const topics_data = await listGroups({
+    type: 'topic',
+  });
+  const location_data = await listGroups({
+    type: 'geography',
+  });
+  const organizations_data = await listOrganizations({
+    input: {
+      detailed: true,
+      includeUsers: true
+    }
+  });
   return {
     props: {
-      csrfToken: await getCsrfToken(context),
+      csrfToken: csrfToken ? csrfToken : "",
+      organizations_data: organizations_data,
+      topics_data: topics_data,
+      location_data: location_data
     },
   };
 }
 
-export default function LoginPage({ csrfToken }: { csrfToken: string }) {
-  const [errorMessage] = useState<string | null>(null);
+export default function LoginPage({
+  csrfToken, 
+  organizations_data, 
```

Comment:
I think it would be better to use snakeCase for variables

Reply:
Resolved.